src/components/client.ts

```diff
@@ -19,17 +19,18 @@ import { MethodCallValidator, MethodNotFoundError } from "@open-rpc/schema-utils
 
 export interface Options {
   transport: {
-    type: "websocket" | "http" | "https";
+    type: "websocket" | "http" | "https" | "postmessagewindow" | "postmessageiframe";
     host: string;
     port: number;
     path?: string;
-  }
+    protocol?: string;
```

Comment:
see comment about protocol. It seems like we could lose this parameter and just do a logical name mapping internally from type => protocol names. Unless protocol is used in some other way.